CURA-3495 Open project/models menu

LipuFei opened the pull request:
Add a dialog asking the user when he/she is opening a Cura project file. He/she can choose to:
1. open as a Cura project, or
2. import models from this project file.
It is possible to set a default option for this. This is configurable in the settings dialog.

Some screenshots:
![image](https://cloud.githubusercontent.com/assets/2630468/24106086/2f9a4942-0d87-11e7-9fc6-32aa03ed29ce.png)
![image](https://cloud.githubusercontent.com/assets/2630468/24141491/e1fd3cf6-0e23-11e7-84a8-75d1d4f968f0.png)

fieldOfView commented on resources/qml/AskOpenAsProjectOrModelsDialog.qml:
```diff
@@ -0,0 +1,126 @@
+        {
+            text: catalog.i18nc("@text:window", "This is a Cura project file. Would you like to open it as a project\nor import the models from it?")
+            anchors.margins: UM.Theme.getSize("default_margin").width
+            font: UM.Theme.getFont("default")
```
This dialog lives in the "system-styled" world (same as the Preferences panes). It should use a Label instead of a Text element, and not specify a font (so it uses the system font).

fieldOfView commented on resources/qml/Preferences/GeneralPage.qml:
```diff
@@ -18,13 +18,25 @@ UM.PreferencesPage
         for(var i = 0; i < languageList.count; i++)
         {
-            if (languageComboBox.model.get(i).code == languageCode)
+            if (languageComboBox.model.get(i).code == code)
```
```code``` is not defined here, ```languageCode``` is

fieldOfView commented on resources/qml/Preferences/GeneralPage.qml:
```diff
@@ -376,6 +391,54 @@ UM.PreferencesPage
+                {
+                    Label
+                    {
+                        text: catalog.i18nc("@window:text", "Default behaviour when opening a project file: ")
```
This needs a ```anchors.verticalCenter: choiceOnOpenProjectDropDownButton.verticalCenter```

fieldOfView commented on resources/qml/Preferences/GeneralPage.qml:
```diff
@@ -376,6 +391,54 @@ UM.PreferencesPage
+                            Component.onCompleted: {
+                                append({ text: catalog.i18nc("@option:openProject", "Always ask"), code: "always_ask" })
+                                append({ text: catalog.i18nc("@option:openProject", "Always open as a project"), code: "open_as_project" })
+                                append({ text: "Always import models", code: "open_as_model" })
```
This option is not translatable

LipuFei commented:
Hi @fieldOfView , all fixed :)

LipuFei commented:
@jackha Fixed some bugs and adjusted the layout for the Preferences dialog.

LipuFei commented:
@jackha Opening a project file from the recent file list now also shows the dialog.

LipuFei commented:
@jackha Ok, should be ready now.
Made a few more fixes:
- fixed for opening from recent file list (should have the same behavior as "open file")
- fixed a few typos and bugs
- fixed a problem that recent file list shows only the files read by ReadMeshJob. The problem is that, if a project file is read as a mesh file, it will appear in the recent file list; if it is read as a project file, it won't.
